py/foundation/in_memory.py

```diff
@@ -217,6 +221,57 @@ def __len__(self) -> int:
   def from_dict(d: Dict[str, Any]) -> 'InMemoryText':
     ...
 
+@attr.s(auto_attribs=True)
+class InMemorySpatialText(Text):
+  _id: str
+  _lines: List[InMemoryText]
+
+  @property
+  def type(self) -> str:
+    return "SpatialText"
+  
+  @property
+  def id(self) -> str:
+    return self._id
+
+  def _get_in_memory_text(self) -> 'InMemoryText':
+    # Gets a one dimensional InMemoryText
+    full_list: List[Union[Word, Whitespace]] = list(self._lines[0].get_children_with_spaces())
+    new_line = InMemoryWhitespace(text='\n')
+    for i in range(1, len(self._lines)):
+      full_list.append(new_line)
+      full_list.extend(self._lines[i].get_children_with_spaces())
+    
+    return InMemoryText('TODO', tuple(full_list))
+  
+  def __str__(self) -> str:
+    return '\n'.join([str(line) for line in self._lines])
+  
+  def __len__(self) -> int:
+    # +1 accounts for new line at each line, -1 because no new line at the end
+    return sum([len(line) + 1 for line in self._lines]) - 1
+  
+  def get_children(self) -> Iterable[Word]:
+    return self._get_in_memory_text().get_children()
+
+  def lstrip(self) -> 'InMemoryText':
+    return self._get_in_memory_text().lstrip()
+  
+  def rstrip(self) -> 'InMemoryText':
+    return self._get_in_memory_text().rstrip()
```

Comment:
`InMemorySpatialText` might not need these strip methods -- do they make sense to keep? Or only have them on `InMemoryText`?

Reply:
I did this because the `Text` interface requires it, and I had InMemorySpacialText extend that. Should I make a new interface for SpatialText, do you think?

Reply:
Got it. Yeah I think SpatialText and Text are two different things. Didn't we decide that it was too complicated to have them all in a unified interface in our chat yesterday? Happy to chat more at 5 about it -- no need to resolve now